Почему класс AccessDeniedException в нашем случае
лучше наследовать от Exception, а не от RuntimeException?

@CodeReview(approved = true)

class AccessDeniedException extends Exception {

}
//исключениe проверяемoе (checked)
//исключения делятся на проверяемые (checked) и непроверяемые (unchecked).
//К проверяемым относятся исключения типа Throwable (а также наследуемые
//от него) и типа Exception (и наследуемые от него, кроме исключений
//типа RuntimeException).